onnxruntime/core/providers/cpu/nn/dropout_op.h

```diff
@@ -65,8 +65,7 @@ Status Dropout<T1, T2, trainable_dropout>::Compute(OpKernelContext* context) con
   ORT_ENFORCE(!mask || mask->Shape() == X_shape, "X and mask should have the same shape");
 
   const Tensor* training_mode = context->Input<Tensor>(2);
-  if ((0 == ratio_value /*Backward compat with TrainableDropout*/) ||
-      !trainable_dropout && (training_mode == nullptr || *(training_mode->Data<bool>()) == false)) {
+  if ((0 == ratio_value) || (training_mode == nullptr || *(training_mode->Data<bool>()) == false)) {
```

Comment:
>0 == ratio_value) [](start = 7, length = 17)

it seems this is for TrainableDropout only, is it?

Reply:
this also applies to Dropout-12 when training_mode is true and ratio is set to zero (unlikely scenario), in which case we should short-circuit the calculation and take this branch.